brewtils/plugin.py

```diff
@@ -1,36 +1,33 @@
 # -*- coding: utf-8 -*-
 
-import json
 import logging
 import logging.config
 import os
-import sys
-from concurrent.futures import ThreadPoolExecutor
-
-import six
 import threading
-from requests import ConnectionError as RequestsConnectionError
 
 import brewtils
 from brewtils.errors import (
+    ConflictError,
+    PluginValidationError,
     ValidationError,
-    RequestProcessingError,
     DiscardMessageException,
-    RepublishRequestException,
-    RestConnectionError,
-    PluginValidationError,
-    RestClientError,
-    parse_exception_as_json,
-    ConflictError,
+    RequestProcessingError,
 )
 from brewtils.log import DEFAULT_LOGGING_CONFIG
-from brewtils.models import Instance, Request, System
+from brewtils.models import Instance, System
 from brewtils.request_consumer import RequestConsumer
+from brewtils.request_handling import EasyRequestUpdater, NoopUpdater, RequestProcessor
 from brewtils.rest.easy_client import EasyClient
 from brewtils.schema_parser import SchemaParser
 
+# This is what enables request nesting to work easily
 request_context = threading.local()
 
+# These are not thread-locals - they should be set in the Plugin __init__ and then never
+# touched. This allows us to do sanity checks when creating nested Requests.
+host = ""
+port = None
```

Comment:
If these are globals, let's make it obvious with `_HOST` and `_PORT`. Ideally people shouldn't be touching these, and since they are common names, we'll avoid accidentally stepping on things.